sources/engine/Stride.Graphics/Direct3D/SwapChainGraphicsPresenter.Direct3D.cs

```diff
@@ -56,13 +60,39 @@ public SwapChainGraphicsPresenter(GraphicsDevice device, PresentationParameters
         {
             PresentInterval = presentationParameters.PresentationInterval;
 
+            flipModelSupport = CheckFlipModelSupport(device);
+            tearingSupport = CheckTearingSupport(device);
+
             // Initialize the swap chain
             swapChain = CreateSwapChain();
 
             backBuffer = new Texture(device).InitializeFromImpl(swapChain.GetBackBuffer<BackBufferResourceType>(0), Description.BackBufferFormat.IsSRgb());
 
             // Reload should get backbuffer from swapchain as well
             //backBufferTexture.Reload = graphicsResource => ((Texture)graphicsResource).Recreate(swapChain.GetBackBuffer<SharpDX.Direct3D11.Texture>(0));
+
+            static bool CheckFlipModelSupport(GraphicsDevice device)
+            {
+                // From https://github.com/walbourn/directx-vs-templates/blob/main/d3d11game_win32_dr/DeviceResources.cpp#L138
+                using var dxgiDevice = device.NativeDevice.QueryInterface<SharpDX.DXGI.Device>();
+                using var dxgiAdapter = dxgiDevice.Adapter;
+                using var dxgiFactory = dxgiAdapter.GetParent<SharpDX.DXGI.Factory4>();
```

Comment:
Technically this bug was introduced in the #1594 pull, but can you wrap this around a try-catch block?
On Win7, this call throws a SharpDXException with `E_NOINTERFACE/No such interface supported`, so should return false on exception (this is because FlipDiscard is only supported on Win10+.

Reply:
Thanks, try/catch has been added.